### ISSUE TITLE
Segmentation model gets wrong  result when enable MKLDNN 

### ISSUE THREAD

**Issue by juncaipeng** [PaddlePaddle/Paddle#35307]
Paddle: develop branch

```
#  Install Paddle

git clone https://github.com/PaddlePaddle/PaddleSeg.git
cd PaddleSeg

# change to develop branch
git checkout -b develop origin/develop 

cd deploy/python

wget https://paddleseg.bj.bcebos.com/dygraph/demo/bisenet_demo_model.tar.gz
tar zxf bisenet_demo_model.tar.gz
wget https://paddleseg.bj.bcebos.com/dygraph/demo/cityscapes_demo.png

python infer.py --config bisenetv2_demo_model/deploy.yaml --image_path cityscapes_demo.png --device cpu --enable_mkldnn False
# the correct result image is saved in ./output

python infer.py --config bisenetv2_demo_model/deploy.yaml --image_path cityscapes_demo.png --device cpu --enable_mkldnn True
# the wrong result image is saved in ./output
```

**Comment by jakpiase** [comment]
This issue was associated with wrong memory tag format deduction. It will be fixed as soon as fixing PR will be merged

**Comment by jakpiase** [comment]
Hi @juncaipeng, could you please check and confirm if this issue is resolved after merging #37310?

**Comment by lidanqing-vv** [comment]
Close this issue cause the fix PR has been merged
